include/RAJA/pattern/launch/launch_core.hpp

```diff
@@ -208,6 +210,26 @@ class LaunchContext
     return static_cast<T*>(mem_ptr);
   }
 
+  template<typename T>
+  RAJA_HOST_DEVICE createMemoryArena(T *ptr, size_t bytes)
+  {
+    void * mem_arena_ptr = static_cast<char*>(&ptr[0]);
+  }
+
+  // TODO handle alignment
+  template<typename T>
+  RAJA_HOST_DEVICE T* getArenaMemory(size_t bytes)
```

Comment:
I don't think `bytes` reflects well how this is used, it seems to be a number of values, not a number of bytes.